pkg/controller/cassandrabackup/reconcile.go

```diff
@@ -0,0 +1,249 @@
+		return reconcile.Result{}, err
+	}
+
+	// Validate the duration if it's set
+	if cassandraBackup.Spec.Duration != "" {
+		if _, err := time.ParseDuration(cassandraBackup.Spec.Duration); err != nil {
+			r.recorder.Event(
+				cassandraBackup,
+				corev1.EventTypeWarning,
+				"BackupFailedDurationParseError",
+				fmt.Sprintf("Duration %s can't be parsed", cassandraBackup.Spec.Duration))
+			return reconcile.Result{}, nil
+		}
+	}
+
+	// Get CassandraCluster object
+	cc := &api.CassandraCluster{}
+	if err := r.client.Get(context.TODO(),
+		types.NamespacedName{Name: cassandraBackup.Spec.CassandraCluster,
+			Namespace: cassandraBackup.Namespace}, cc); err != nil {
+		if k8sErrors.IsNotFound(err) {
+			r.recorder.Event(
+				cassandraBackup,
+				corev1.EventTypeWarning,
+				"CassandraClusterNotFound",
+				fmt.Sprintf("Datacenter %s of cluster %s to backup not found", cassandraBackup.Spec.Datacenter, cassandraBackup.Spec.CassandraCluster))
+
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	if cassandraBackup.IsScheduled() { // Add or update a cron task to handle the backup
+
+		if err := cassandraBackup.Spec.ValidateScheduleFormat(); err != nil {
+			r.recorder.Event(
+				cassandraBackup,
+				corev1.EventTypeWarning,
+				"BackupInvalidSchedule",
+				fmt.Sprintf("Schedule %s is not valid: %s", cassandraBackup.Spec.Schedule, err.Error()))
+
+			// If schedule is invalid and the object already existed, we reset it to what it was
+			if instanceChanged {
+				logrus.WithFields(logrus.Fields{"backup": request.NamespacedName}).Info(undoScheduleChange)
+				//We retrieved our last-applied-configuration stored in the object
+				var oldCassandraBackup api.CassandraBackup
+				err := json.Unmarshal([]byte(cassandraBackup.Annotations[annotationLastApplied]), &oldCassandraBackup)
+				if err != nil {
+					logrus.WithFields(logrus.Fields{"backup": request.NamespacedName,
+						"err": err}).Error(retryFailedUndoSchedule)
+					return reconcile.Result{}, err
+				}
+
+				cassandraBackup.Spec.Schedule = oldCassandraBackup.Spec.Schedule
+				cassandraBackup.Annotations[annotationLastApplied], _ = cassandraBackup.ComputeLastAppliedAnnotation()
+				return reconcile.Result{}, err
+			}
+
+			// The schedule is invalid and the object is new, we stop here
+			return reconcile.Result{}, nil
+		}
+
+		if skipped, err := r.scheduler.AddOrUpdate(
+			cassandraBackup, func() { r.backupData(cassandraBackup, cc, reqLogger) }, &r.recorder); err != nil {
+			r.recorder.Event(cassandraBackup, corev1.EventTypeWarning, "BackupScheduleError",
+				fmt.Sprintf("Wasn't able to schedule job %s: %s", cassandraBackup.Name, err.Error()))
+		} else if skipped {
+			return reconcile.Result{}, nil
+		}
+
+		// Add Finalizer
+		r.setFinalizer(cassandraBackup, true)
+	}
+
+	if cassandraBackup.IsScheduled() {
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, r.backupData(cassandraBackup, cc, reqLogger)
```

Comment:
What happens if we apply changes to a CassandraBackup and a backup is already running ? 
Is it starting another backup ? Does it kill the one who is running ? Does it do nothing ?

Reply:
@erdrix If the backup is scheduled, it's updated. If it's not, no new backup runs https://github.com/Orange-OpenSource/casskop/blob/backup-operator/pkg/controller/cassandrabackup/reconcile.go#L120-L121